ALICE3/TableProducer/onTheFlyTOFPID.cxx

```diff
@@ -0,0 +1,274 @@
+
+  o2::base::Propagator::MatCorrType matCorr = o2::base::Propagator::MatCorrType::USEMatCorrNONE;
+
+  // needed: random number generator for smearing
+  TRandom3 lPRNG;
+
+  // for handling basic QA histograms if requested
+  HistogramRegistry histos{"Histos", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  void init(o2::framework::InitContext& initContext)
+  {
+    lPRNG.SetSeed(0); // fully randomize
+
+    if (doQAplots) {
+      const AxisSpec axisMomentum{static_cast<int>(80), 0.0f, +4.0f, "#it{p} (GeV/#it{c})"};
+      const AxisSpec axisVelocity{static_cast<int>(110), 0.0f, +1.1f, "Measured #beta"};
+      histos.add("h2dVelocityVsMomentumInner", "h2dVelocityVsMomentumInner", kTH2F, {axisMomentum, axisVelocity});
+      histos.add("h2dVelocityVsMomentumOuter", "h2dVelocityVsMomentumOuter", kTH2F, {axisMomentum, axisVelocity});
+    }
+  }
+
+  template <typename mcParticleType>
+  void convertMCParticleToO2Track(mcParticleType& particle, o2::track::TrackParCov& o2track)
+  {
+    // FIXME: this is a fundamentally important piece of code.
+    // It could be placed in a utility file instead of here.
+    std::array<float, 3> xyz = {static_cast<float>(particle.vx()), static_cast<float>(particle.vy()), static_cast<float>(particle.vz())};
+    std::array<float, 3> ptetaphi = {static_cast<float>(particle.pt()), static_cast<float>(particle.eta()), static_cast<float>(particle.phi())};
+    auto pdgInfo = pdg->GetParticle(particle.pdgCode());
+    int charge = 0;
+    if (pdgInfo != nullptr)
+      charge = pdgInfo->Charge();
+    std::array<float, 5> params;
+    std::array<float, 15> covm = {0.};
+    float s, c, x;
+    o2::math_utils::sincos(ptetaphi[2], s, c);
+    o2::math_utils::rotateZInv(xyz[0], xyz[1], x, params[0], s, c);
+    params[1] = xyz[2];
+    params[2] = 0.; // since alpha = phi
+    auto theta = 2. * std::atan(std::exp(-ptetaphi[1]));
+    params[3] = 1. / std::tan(theta);
+    params[4] = charge / ptetaphi[0];
+
+    // Initialize TrackParCov in-place
+    new (&o2track)(o2::track::TrackParCov)(x, ptetaphi[2], params, covm);
+  }
+
+  float trackLength(o2::track::TrackParCov track, float lX0, float lX1, float lMagneticField)
+  {
+    // utility class to calculate track length from track position X0 to track position X1
+    // with a given magnetic field
+    std::array<float, 3> lPointN;
+    std::array<float, 3> lPointNplus;
+    float lLength = 0.0;
+    track.propagateTo(lX0, lMagneticField);
+    for (int iStep = 1; iStep < lNStepsLIntegrator; iStep++) {
+      track.getXYZGlo(lPointN);
+      float lPosition = lX0 + (lX1 - lX0) * (static_cast<float>(iStep)) / (static_cast<float>(lNStepsLIntegrator - 1));
+      track.propagateTo(lPosition, lMagneticField);
+      track.getXYZGlo(lPointNplus);
+      lLength += std::hypot(lPointNplus[0] - lPointN[0], lPointNplus[1] - lPointN[1], lPointNplus[2] - lPointN[2]);
+    }
+    return lLength;
+  }
+
+  float Velocity(float lMomentum, float lMass)
```

Comment:
Same comment about the `l`.

Reply:
no 'l' anywhere anymore in latest version (commit soon)